File: src/main/java/com/leafchild/springmvc/service/UserService.java
package com.leafchild.springmvc.service;

import com.leafchild.springmvc.model.User;

import java.util.List;


/**
 * Created by: leaf
 * Project: spring-mvc-angular-test
 * Date: 11/26/15
 * Time: 12:53 PM
 */

public interface UserService {

	User findOne(long id);

	User findByName(String name);

	User save(User user);

	void update(User user);

	void delete(long id);

	List<User> findAll();

	void deleteAll();

	boolean exists(long userId);

	boolean isDummyUser(User user);

}

File: src/main/webapp/static/js/service/user_service.js
/**
 * Created by: leaf
 * Date: 11/26/15
 * Time: 12:33 PM
 */
'use strict';

App.service('userService', ['$resource', 'baseUrl', function($resource, baseUrl){

	this.users = function() {
		return $resource(baseUrl + 'user', null,
		  {
			  'update': {method: 'PUT'},
			  'getAll': {method: 'GET', isArray: true}
		  });
	};

}]);